include/crm/common/output.h

```diff
@@ -369,6 +369,35 @@ struct pcmk__output_s {
      * \param[in,out] out The output functions structure.
      */
     void (*end_list) (pcmk__output_t *out);
+
+    /*!
+     * \internal
+     * \brief Format a name/value pair.
+     *
+     * Units can optionally be provided for the value.  Note that unlike most
+     * formatting functions, this one returns the formatted string.  It is
+     * assumed that the most common use of this function will be to build up
+     * a string to be output as part of other functions.
+     *
+     * \param[in,out] out   The output functions structure.
+     * \param[in]     name  The name of the nvpair.
+     * \param[in]     value The value of the nvpair.
+     * \param[in]     units Optional units for the value, or NULL.
```

Comment:
They could be.  There's two real differences among the various versions of the functions: The XML version puts quotes around things (but we could just do that globally and everyone gets over it), and the none version returns empty strings.  I don't quite know how to deal with that.  I worry that other things like this might come up in the future.

Reply:
I think always using quotes would be fine (and probably even makes sense for the other formats, when values have spaces). The none formatter won't be calling the function, so I don't think it matters.

If you do make it a utility function, it would make sense to go in nvpair.c